core/model/modx/modprocessor.class.php

```diff
@@ -1408,8 +1408,10 @@ public function cache() {
      */
     public function download() {
         $file = $this->object->get($this->nameField).'.xml';
+        $f = $this->modx->getOption('core_path', null, MODX_CORE_PATH) . 'export/' . $this->objectType . '/' . $file;
```

Comment:
$f variable not clear. Best variant user $fileName instead $file, and $file instead $f.